### PR TITLE
Add ast_dump option: AST_DUMP_EXCLUDE_DOC

### PR DESCRIPTION
**Pull request by fan-y-mishiba2**
Related to #191

### CODE REVIEW

File: tests/ast_dump_with_exclude_doc.phpt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+--TEST--
+ast_dump() with AST_DUMP_EXCLUDE_DOC
+--SKIPIF--
+<?php
+if (!extension_loaded("ast")) print "skip";
+if (PHP_VERSION_ID >= 70000 && PHP_VERSION_ID < 70018) print "skip buggy PHP version";
+if (PHP_VERSION_ID >= 70100 && PHP_VERSION_ID < 70104) print "skip buggy PHP version";
```
Comment (nikic) [inline tests/ast_dump_with_exclude_doc.phpt:7]: These "skip buggy php version"s should not be needed for this test.

File: tests/ast_dump_with_exclude_doc.phpt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * @return void
+ */
+function
+test
+($foo)
+{
+    var_dump(
+        $foo
+    );
+}
```
Comment (nikic) [inline tests/ast_dump_with_exclude_doc.phpt:28]: You can simplify this test by only making it:
```
<?php
/**
 * test
 * @param mixed $foo
 * @return void
 */
function test($foo) {
}
```
The odd formatting was only needed to test an issue with line numbers.

File: tests/ast_dump_with_exclude_doc.phpt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+PHP;
+
+$ast = ast\parse_code($code, $version=50);
```
Comment (nikic) [inline tests/ast_dump_with_exclude_doc.phpt:31]: Please use `$version=80` here to make this test longer lift. I don't think it affects the output.

File: util.php [hunk 4]
```diff
@@ -3,6 +3,7 @@
 use ast\flags;
 
 const AST_DUMP_LINENOS = 1;
+const AST_DUMP_EXCLUDE_DOC = 2;
```
Comment (nikic) [inline util.php:6]: I would prefer to call this `AST_DUMP_EXCLUDE_DOC_COMMENT`.

File: tests/ast_dump_with_exclude_doc.phpt (hunk 1 above)
Comment (fan-y-mishiba2) [reply tests/ast_dump_with_exclude_doc.phpt]: excluded SKIPIF
a93bcb2

File: tests/ast_dump_with_exclude_doc.phpt (hunk 2 above)
Comment (fan-y-mishiba2) [reply tests/ast_dump_with_exclude_doc.phpt:16]: Simplified code
acabd2a

Missing correction of EXPECT
b5e1e3d

File: tests/ast_dump_with_exclude_doc.phpt (hunk 3 above)
Comment (fan-y-mishiba2) [reply tests/ast_dump_with_exclude_doc.phpt]: AST-version set to 80
3191a4e

File: util.php (hunk 4 above)
Comment (fan-y-mishiba2) [reply util.php]: Changed the constant name
5cbf249

**Review by nikic — APPROVED** [review]

**Pull request merged**